document-store/build.gradle.kts

```diff
@@ -16,7 +16,7 @@ plugins {
 
 dependencies {
   api("com.typesafe:config:1.3.2")
-  implementation("org.mongodb:mongo-java-driver:3.12.0")
+  implementation("org.mongodb:mongodb-driver-sync:4.1.1")
```

Comment:
Was this async or have we been using sync mongo?

Reply:
We have always been using sync version, which is now their legacy driver. Starting 3.x, they separated the sync and reactive java drivers.

Reply:
We should probably switch to reactive at some point

Reply:
Interesting proposal to have async for data persistence. I would like to understand how we're going to handle data inconsistency, error handling, and our SLA (is eventual consistency acceptable) with having async data persistence. But, I think it's outside the scope of this review.

Reply:
Async is about data transfer not persistence. The client would be async.